packages/components/src/components/hive-navigation/index.tsx

```diff
@@ -0,0 +1,562 @@
+  const Search = themeConfig?.search?.component;
+
+  const isHive = productName === 'Hive';
+
+  const containerRef = useRef<HTMLDivElement>(null);
+
+  return (
+    <div
+      ref={containerRef}
+      className={cn(
+        'sticky top-0 z-20 border-b border-beige-400/[var(--border-opacity)] bg-[rgb(var(--nextra-bg))] px-6 py-4 text-green-1000 transition-[border-color] duration-500 md:mb-[7px] md:mt-2 dark:border-neutral-700/[var(--border-opacity)] dark:text-neutral-200 [&.light]:border-beige-400/[var(--border-opacity)] [&.light]:bg-white [&.light]:text-green-1000',
+        className?.includes('light') && 'light',
+      )}
+      style={{ '--border-opacity': 0 }}
+    >
+      <TopOfSiteMarker
+        onChange={scrolled => {
+          const container = containerRef.current;
+          if (container) {
+            container.style.setProperty('--border-opacity', scrolled ? '1' : '0');
+          }
+        }}
+      />
+
+      {/* mobile menu */}
+      <div className="flex items-center justify-between md:hidden">
+        <HiveLogoLink />
+        <HamburgerButton />
+      </div>
+
+      {/* desktop menu */}
+      <NavigationMenu className={cn('mx-auto hidden md:flex', className)} delayDuration={0}>
+        <HiveLogoLink />
+        <NavigationMenuList className="bg-white dark:bg-transparent [@media(min-width:1180px)]:ml-16">
+          <NavigationMenuItem>
+            <NavigationMenuTrigger>Products</NavigationMenuTrigger>
+            <NavigationMenuContent>
+              <ProductsMenu isHive={isHive} />
+            </NavigationMenuContent>
+          </NavigationMenuItem>
+          <NavigationMenuItem>
+            <NavigationMenuTrigger>Developer</NavigationMenuTrigger>
+            <NavigationMenuContent>
+              <DeveloperMenu isHive={isHive} />
+            </NavigationMenuContent>
+          </NavigationMenuItem>
+          {!ENTERPRISE_MENU_HIDDEN && (
+            <NavigationMenuItem>
+              <NavigationMenuTrigger>Enterprise</NavigationMenuTrigger>
+              <NavigationMenuContent>
+                <EnterpriseMenu />
+              </NavigationMenuContent>
+            </NavigationMenuItem>
+          )}
+          <NavigationMenuItem>
+            <NavigationMenuTrigger>Company</NavigationMenuTrigger>
+            <NavigationMenuContent>
+              <CompanyMenu>{companyMenuChildren}</CompanyMenu>
+            </NavigationMenuContent>
+          </NavigationMenuItem>
+          <NavigationMenuItem>
+            <NavigationMenuLink
+              href={isHive ? '/#pricing' : 'https://the-guild.dev/graphql/hive#pricing'}
+              className="font-medium"
+            >
+              Pricing
+            </NavigationMenuLink>
+          </NavigationMenuItem>
+        </NavigationMenuList>
+        <div className="flex-1" />
+
+        {children}
+
+        {renderSlot(Search, {
+          // The && and :is(x) selector bump the specificity to 0-2-2 to override Nextra styles.
```

Comment:
please don't do this, it's too overcomplicates all the things, it will be broken and difficult to maintain